aplib.net-demo/Assets/Testing/AplibTests/RangedEnemyTest.cs

```diff
@@ -0,0 +1,139 @@
+using Aplib;
+using Aplib.Core;
+using Aplib.Core.Belief;
+using Aplib.Core.Desire;
+using Aplib.Core.Desire.Goals;
+using Aplib.Core.Intent.Actions;
+using Aplib.Core.Intent.Tactics;
+using Aplib.Integrations.Unity.Actions;
+using NUnit.Framework;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+
+namespace Testing.AplibTests
+{
+    public class RangedEnemyBeliefSet : BeliefSet
+    {
+        public static readonly int FramesToRemember = 300;
+
+        /// <summary>
+        /// The player object in the scene.
+        /// </summary>
+        public Belief<GameObject, GameObject> Player = new(reference: GameObject.Find("Player"), x => x);
+
+        /// <summary>
+        /// The target position for the player to move to.
+        /// Also remembers the target position from a certain amount of frames ago.
+        /// </summary>
+        public MemoryBelief<GameObject, Vector3> TargetPosition = new(GameObject.Find("Ranged Enemy Body"), x => x.transform.position, FramesToRemember);
+
+        /// <summary>
+        /// The respawn position for the player.
+        /// </summary>
+        public Belief<GameObject, Vector3> RespawnPosition = new(GameObject.Find("RespawnPoint"), x => x.transform.position);
+    }
+
+    /// <summary>
+    /// A basic test for the ranged enemy.
+    /// The player will first move within vision range of the enemy.
+    /// The enemy should then walk around the wall to the player and kill him.
+    /// The player then respawns out of vision range of the enemy.
+    /// If the enemy does not attack him then, the test is successful.
+    /// </summary>
+    public class RangedEnemyAplibTest
+    {
+        private float _targetStoppingDistance = 7f;
+        private string _sceneName = "RangedEnemyTestScene";
+
+        [SetUp]
+        public void SetUp()
+        {
+            Debug.Log($"Starting {nameof(RangedEnemyAplibTest)}");
+            SceneManager.LoadScene(_sceneName);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            Debug.Log($"Finished {nameof(RangedEnemyAplibTest)}");
+            SceneManager.UnloadSceneAsync(_sceneName);
+        }
+
+        [UnityTest]
+        public IEnumerator PerformRangedEnemyTest()
+        {
+            InputManager.Instance.enabled = false;
+            RangedEnemyBeliefSet beliefSet = new();
```

Comment:
To prevent conflicts with my other suggestion.
```suggestion
            RangedEnemyBeliefSet rangedEnemyBeliefSet = new();
```